python/lambda-ddb-mysql-etl-pipeline/lambda/dbwrite.py

```diff
@@ -0,0 +1,200 @@
+import json
+import os
+import decimal
+import mimetypes
+import boto3
+import logging
+import pymysql 
+import pandas
+
+logger = logging.getLogger()
+logger.setLevel(logging.DEBUG)
+
+ddb = boto3.resource('dynamodb')
+s3 = boto3.client('s3')
+ddb_table = ddb.Table(os.environ['ASTEROIDS_TABLE'])
+_lambda = boto3.client('lambda')
+db_conn = None
+SCHEMA = os.environ['SCHEMA']
+
+def handler(event, context):
+    """Retrieve s3 object and write to dynamo or mysql table(not-functional in this example, thus comments)
+
+    Args:
+        event (dict, list, str, int, float, NoneType): Provides event specific data to the handler. It
+            can also be of type list, str, int, float, or NoneType.
+        context (obj): Provides methods and properties that contain invocation, function and
+            execution environment information.
+    Raises:
+        Exception: If file format is not of .json or .csv file types.
+
+    Notes:
+        A call to `write_to_mysql()` will require that you setup a DB Connection
+        and set `db_conn` herein to `True`. Otherwise, the current example will
+        only print results to CloudWatch Logs. Pandas & PyMySQL archives are
+        included as layers in the project.
+    """
+    s3_key=event['Records'][0]['s3']['object']['key']
+    stream_obj=s3.get_object(Bucket=os.environ['S3_BUCKET'], Key=s3_key)
+    data = stream_obj['Body'].read().decode('utf-8')
+    if mimetypes.guess_type(s3_key)[0] == "application/json":
+        asteroid_list=json.loads(data, parse_float=decimal.Decimal)
+        write_to_ddb(asteroid_list)
+    elif mimetypes.guess_type(s3_key)[0] ==  "text/csv":
+        write_to_mysql(data, 'asteroids')
+    else:
+        raise Exception("FILE FORMAT NOT RECOGNIZED")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.